src/v2/guide/reactivity.md

```diff
@@ -4,80 +4,80 @@ type: guide
+Đôi khi bạn có thể muốn gán một số thuộc tính cho một đối tượng hiện có, ví dụ bằng cách sử dụng `Object.assign()` hoặc `_.extend()`. Tuy nhiên, các thuộc tính mới được thêm vào này sẽ không kích hoạt thay đổi (tức là khi chúng thay đổi Vue sẽ không re-render). Trong những trường hợp như thế, bạn nên tạo ra một đối tượng mới với các thuộc tính từ cả đối tượng gốc lẫn đối tượng có chứa các thuộc tính bạn muốn thêm vào, ví dụ:
 
 ``` js
-// instead of `Object.assign(this.someObject, { a: 1, b: 2 })`
+// Không nên `Object.assign(this.someObject, { a: 1, b: 2 })`
 this.someObject = Object.assign({}, this.someObject, { a: 1, b: 2 })
 ```
 
-There are also a few array-related caveats, which were discussed earlier in the [list rendering section](list.html#Caveats).
+Đồng thời cũng vẫn còn tồn tại một số hạn chế liên quan đến Array, những điều đó chúng ta sẽ cùng thảo luộn ở [Render danh sách](list.html#Caveats).
 
-## Declaring Reactive Properties
+## Khai báo thuộc tính Reactivity
 
-Since Vue doesn't allow dynamically adding root-level reactive properties, you have to initialize Vue instances by declaring all root-level reactive data properties upfront, even with an empty value:
+Vì Vue không cho phép thêm động các thuộc tính phản ứng mức root, nên bạn phải khởi tạo các thực thể Vue bằng cách khai báo trước tất cả các thuộc tính dữ liệu reactive ở root, thậm chí với một giá trị rỗng:
 
 ``` js
 var vm = new Vue({
   data: {
-    // declare message with an empty value
+    // khai báo message với giá trị rỗng
     message: ''
   },
   template: '<div>{{ message }}</div>'
 })
-// set `message` later
+// thiết lập sau giá trị cho `message`
 vm.message = 'Hello!'
 ```
 
-If you don't declare `message` in the data option, Vue will warn you that the render function is trying to access a property that doesn't exist.
+Nếu bạn không khai báo `message` ở trong `data`, Vue sẽ cảnh báo bạn rằng `render function` đang cố gắng truy cập một thuộc tính không tồn tại.
 
-There are technical reasons behind this restriction - it eliminates a class of edge cases in the dependency tracking system, and also makes Vue instances play nicer with type checking systems. But there is also an important consideration in terms of code maintainability: the `data` object is like the schema for your component's state. Declaring all reactive properties upfront makes the component code easier to understand when revisited later or read by another developer.
+Có những lý do kỹ thuật đằng sau hạn chế này - nó loại bỏ một lớp các trường hợp ngoại lệ trong hệ thống theo dõi phụ thuộc, và cũng làm cho các cá thể Vue hoạt động một cách tốt hơn với các hệ thống kiểm tra kiểu. Nhưng cũng có một cân nhắc quan trọng về khả năng bảo trì mã nguồn (code): đối tượng `data` giống như lược đồ cho trạng thái của các components. Khai báo trước tất cả các thuộc tính `reactive` làm cho code của bạn dễ hiểu hơn khi được xem lại sau hoặc được đọc bởi một nhà phát triển khác.
```

Comment:
* "schema" ở đây giống như là cấu trúc hơn là lược đồ
* "hoạt động tốt hơn" thay vì "hoạt động một cách tốt hơn"
* Câu "Khai báo trước tất cả các thuộc tính `reactive` làm cho code của bạn dễ hiểu hơn khi được xem lại sau hoặc được đọc bởi một nhà phát triển khác." khá lủng củng. Có thể dịch thoát ý ngắn gọn thành: "Việc khai báo trước toàn bộ các thuộc tính reactive giúp code của bạn trở nên dễ đọc, dễ hiểu hơn." (với ai thì cũng thế).